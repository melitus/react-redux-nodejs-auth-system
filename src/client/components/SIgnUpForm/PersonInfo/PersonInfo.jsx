// @flow
import React from 'react';
import { Field, FormSection } from 'redux-form';

import Input from '../../Field/index';

const PersonInfo = ({ firstName, lastName }) => (
  <FormSection name="personInfo">
    <label target="signup.personInfo.firstName">First Name</label>
    <Field
      id="signup.personInfo.firstName"
      name="firstName"
      component={Input}
      placeholder="First Name"
      value={firstName}
      autoFocus
    />

    <label target="signup.personInfo.lastName">Last Name</label>
    <Field
      id="signup.personInfo.lastName"
      name="lastName"
      component={Input}
      placeholder="Last Name"
      value={lastName}
    />
  </FormSection>
);

PersonInfo.defaultProps = {
  firstName: '',
  lastName: '',
};

export default PersonInfo;
