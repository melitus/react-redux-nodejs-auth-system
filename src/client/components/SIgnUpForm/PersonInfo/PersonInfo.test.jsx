import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import PersonInfo from './PersonInfo';

enzyme.configure({ adapter: new Adapter() });

describe('<PersonInfo />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <PersonInfo />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
