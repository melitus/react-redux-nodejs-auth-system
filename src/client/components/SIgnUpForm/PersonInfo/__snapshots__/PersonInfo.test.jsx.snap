exports[`<PersonInfo /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PersonInfo
    firstName=""
    lastName="" />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label
          target="signup.personInfo.firstName">
          First Name
</label>,
        <Field
          autoFocus={true}
          component={undefined}
          id="signup.personInfo.firstName"
          name="firstName"
          placeholder="First Name"
          value="" />,
        <label
          target="signup.personInfo.lastName">
          Last Name
</label>,
        <Field
          component={undefined}
          id="signup.personInfo.lastName"
          name="lastName"
          placeholder="Last Name"
          value="" />,
      ],
      "component": "div",
      "name": "personInfo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "First Name",
          "target": "signup.personInfo.firstName",
        },
        "ref": null,
        "rendered": "First Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": true,
          "component": undefined,
          "id": "signup.personInfo.firstName",
          "name": "firstName",
          "placeholder": "First Name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Last Name",
          "target": "signup.personInfo.lastName",
        },
        "ref": null,
        "rendered": "Last Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "component": undefined,
          "id": "signup.personInfo.lastName",
          "name": "lastName",
          "placeholder": "Last Name",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label
            target="signup.personInfo.firstName">
            First Name
</label>,
          <Field
            autoFocus={true}
            component={undefined}
            id="signup.personInfo.firstName"
            name="firstName"
            placeholder="First Name"
            value="" />,
          <label
            target="signup.personInfo.lastName">
            Last Name
</label>,
          <Field
            component={undefined}
            id="signup.personInfo.lastName"
            name="lastName"
            placeholder="Last Name"
            value="" />,
        ],
        "component": "div",
        "name": "personInfo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "First Name",
            "target": "signup.personInfo.firstName",
          },
          "ref": null,
          "rendered": "First Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": true,
            "component": undefined,
            "id": "signup.personInfo.firstName",
            "name": "firstName",
            "placeholder": "First Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Last Name",
            "target": "signup.personInfo.lastName",
          },
          "ref": null,
          "rendered": "Last Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": undefined,
            "id": "signup.personInfo.lastName",
            "name": "lastName",
            "placeholder": "Last Name",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
